packages/contracts/test/framework/plugin/plugin-setup-processor.ts

```diff
@@ -427,135 +540,232 @@ describe.skip('Plugin Setup Processor', function () {
+          targetDao.address,
+          pluginRepoPointer,
+          EMPTY_DATA
+        );
 
+        const pluginInstallationId = getPluginInstallationId(
+          targetDao.address,
+          plugin
+        );
+        // Check that both setupId are valid at this moment as none of them have been applied yet.
         await expect(
-          psp.applyInstallation(
+          psp.validatePreparedSetupId(
+            pluginInstallationId,
+            firstPreparedSetupId
+          )
+        ).not.to.be.reverted;
+        await expect(
+          psp.validatePreparedSetupId(
+            pluginInstallationId,
+            secondPreparedSetupId
+          )
+        ).not.to.be.reverted;
+        await expect(
+          psp.callStatic.applyInstallation(
             targetDao.address,
-            setupUV1.address,
-            repoU.address,
-            unpreparedPlugin,
-            unPreparedPermissions
+            createApplyInstallationParams(
+              plugin,
+              pluginRepoPointer,
+              firstPreparedPermissions,
+              firstPreparedHelpers
+            )
           )
-        ).to.be.revertedWithCustomError(psp, 'SetupNotPrepared');
-      });
+        ).not.to.be.reverted;
+        await expect(
+          psp.callStatic.applyInstallation(
+            targetDao.address,
+            createApplyInstallationParams(
+              plugin,
+              pluginRepoPointer,
+              secondPreparedPermissions,
+              secondPreparedHelpers
+            )
+          )
+        ).not.to.be.reverted;
 
-      it('applies a prepared installation', async () => {
-        const {plugin, permissions: permissions} = await prepareInstallation(
+        // Lets install one of them.
+        await applyInstallation(
           psp,
           targetDao.address,
-          setupUV1.address,
-          repoU.address,
-          EMPTY_DATA
+          plugin,
+          pluginRepoPointer,
+          firstPreparedPermissions,
+          firstPreparedHelpers
         );
 
         await expect(
-          psp.applyInstallation(
+          psp.validatePreparedSetupId(
+            pluginInstallationId,
+            firstPreparedSetupId
+          )
+        ).to.be.reverted;
+        await expect(
+          psp.validatePreparedSetupId(
+            pluginInstallationId,
+            secondPreparedSetupId
+          )
+        ).to.be.reverted;
+
+        await expect(
+          psp.callStatic.applyInstallation(
             targetDao.address,
-            setupUV1.address,
-            repoU.address,
-            plugin,
-            permissions
+            createApplyInstallationParams(
+              plugin,
+              pluginRepoPointer,
+              firstPreparedPermissions,
+              firstPreparedHelpers
+            )
           )
-        )
-          .to.emit(psp, EVENTS.InstallationApplied)
-          .withArgs(targetDao.address, plugin);
-      });
+        ).to.be.reverted;
```

Comment:
add at least one non-static call and check for the revert `PluginAlreadyInstalled` so that we can verify we go into the right code path.

Reply:
yep, removed staticcall.